poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -679,7 +710,7 @@ public Asset fillLocation(Sensor sensor, Map<Long, Rack> racksMap, Map<Long, Row
       return sensors;
    }
 
-   public List<Asset> filterAsset(Map<String, Asset> exsitingAsset, List<Asset> assetsFromPowerIQ) {
+   public List<Asset> filterSensorAsset(Map<String, Asset> exsitingAsset, List<Asset> assetsFromPowerIQ) {
       if (assetsFromPowerIQ == null || assetsFromPowerIQ.isEmpty()) {
          return null;
       } else if (exsitingAsset == null || exsitingAsset.isEmpty()) {
```

Comment:
check line 
for(Asset asset: assetFromPowerIQ)
String sensorID = asset.getJustificationfields().get(FlowgateConstant.SENSOR_ID_FROM_POWERIQ);
         if (sensorID == null) {

since the data is read from PowerIQ directly , the sensor id should always empty. Please double check if your code could be executed correctly.